components/nuclio/deploy/component.yaml

```diff
@@ -0,0 +1,28 @@
+name: nuclio deploy
+description: auto build and deploy nuclio function.
+inputs:
+  - {name: Url, type: String, description: 'url/path to source code, zip archive, git path, or notebook'}
+  - {name: Name, type: String, description: 'function name'}
+  - {name: Project, type: String, description: 'project name', default: 'default'}
+  - {name: Tag, type: String, description: 'function version tag', default: ''}
+  - {name: Dashboard, type: String, description: 'nuclio dashboard service url', default: 'http://nuclio-dashboard.nuclio:8070'}
+  - {name: Spec, type: String, description: 'override function spec, Json {key: value, ..}', default: ''}
+  - {name: Env, type: String, description: 'override function env var, Json {key: value, ..}', default: ''}
+  - {name: Mount, type: String, description: 'volume mount, [vol-type:]<vol-url>:<dst-path>', default: ''}
+outputs:
+  - {name: address, type: String, description: 'function endpoint url'}
```

Comment:
Maybe rename `address` to `Endpoint` (in two places)?

Reply:
What does a function 'deployment' mean? i believe this is more like 'execute' a function?

Reply:
@Ark-kun renamed

@animeshsingh `deploy` takes code + extra files + spec (from code, notebook, zip, git, ..), builds a container, and creates the CRD/deployment for it. a function may be invoked independently e.g. when its doing model serving or ETL from a Kafka/pubsub topic.

Reply:
see more details/examples on how deploy works in: https://github.com/nuclio/nuclio-jupyter